### ISSUE TITLE
Actor killed after training one episode.

### ISSUE THREAD

**Issue by zhanghaoie** [OpenLLMAI/OpenRLHF#187]
Hi, I have found a strange issue. After collect all experiences, the actor and critic start to train, but as there are 4 actors and 2 critics, the actors finished training first, as it has no activities and the high memory usage, it may killed by the SIGKILL. Is there a method to solve this issue? 

following is the message:
The actor is dead because its worker process has died. Worker exit type: SYSTEM_ERROR Worker exit detail: Worker unexpectedly exits with a connection error code 2. End of file. There are some potential root causes. (1) The process is killed by SIGKILL by OOM killer due to high memory usage. (2) ray stop --force is called. (3) The worker is crashed unexpectedly due to SIGSEGV or other unexpected errors.

**Comment by wuxibin89** [comment]
@zhanghaoie Normally, after actors finished training, the last step is to save actor model to Hugging Face pretrained model.

**Comment by wuxibin89** [comment]
Can you post training scripts and hardware info：GPU、CPU、MEM？

**Comment by zhanghaoie** [comment]
GPU: 8*A100, MEM: 1T, used 537GB， but The docker requests 500G(max memory can be requested)
The training script:

#!/bin/bash
set +x

#REWARD_MODEL="IDEA-CCNL/Ziya-LLaMA-7B-Reward"
#SFT_MODEL="meta-llama/Llama-2-7b-chat-hf"

REWARD_MODEL="/data/oss_bucket_0/rlhf/sft"
SFT_MODEL="/data/oss_bucket_0/rlhf/sft"
DATASET="/data/oss_bucket_0/dataset/full-hh-rlhf/,/data/oss_bucket_0/dataset/oasst1/"

#workdir=$(cd $(dirname $0); pwd)
#
#echo ${workdir}
#parent_dir=$(dirname "$(pwd)")
#WORK_DIR=$(dirname "$parent_dir")
WORK_DIR="$(pwd)"

echo ${WORK_DIR}

RUNTIME_ENV_JSON="{\"working_dir\": \"${WORK_DIR}\"}"
SAVE_PATH="${WORK_DIR}/test/llama"


/root/.local/bin/ray job submit --address="http://127.0.0.1:8265" \
    -- python3 examples/train_ppo_ray.py \
    --ref_num_nodes 1 \
    --ref_num_gpus_per_node 1 \
    --reward_num_nodes 1 \
    --reward_num_gpus_per_node 1 \
    --critic_num_nodes 1 \
    --critic_num_gpus_per_node 2 \
    --actor_num_nodes 1 \
    --actor_num_gpus_per_node 4 \
    --pretrain "${SFT_MODEL}" \
    --reward_pretrain "${REWARD_MODEL}" \
    --save_path "${SAVE_PATH}" \
    --micro_train_batch_size 1 \
    --train_batch_size 128 \
    --micro_rollout_batch_size 2 \
    --rollout_batch_size 1024 \
    --max_epochs 1 \
    --prompt_max_len 1024 \
    --generate_max_len 1024 \
    --zero_stage 2 \
    --bf16 \
    --actor_learning_rate 5e-7 \
    --critic_learning_rate 9e-6 \
    --inference_tp_size 1 \
    --init_kl_coef 0.01 \
    --prompt_data "${DATASET}" \
    --prompt_data_probs 0.6,0.4 \
    --max_samples 2000 \
    --normalize_reward \
    --actor_init_on_gpu \
    --adam_offload \
    --flash_attn \
    --gradient_checkpointing

**Comment by hijkzzz** [comment]
Are you running a model of 13B

**Comment by zhanghaoie** [comment]
> Are you running a model of 13B

yeah, I have used Baichuan 13b model for actors and critics.

**Comment by zhanghaoie** [comment]
> ```python
>     ray.get(refs)
> ```

ray.get(refs) incurred the error.

**Comment by hijkzzz** [comment]
I suspect that  OOM caused it  @zhanghaoie

**Comment by hijkzzz** [comment]
@zhanghaoie you could try [--grad_accum_dtype", type=str, default=None)]  grad_accum_dtype "bf16" to save cpu memory

**Comment by zhanghaoie** [comment]
> @zhanghaoie you could try [--grad_accum_dtype", type=str, default=None)] grad_accum_dtype "bf16" to save cpu memory

Thank you for you advice! I will try it.
